report/sections/Evaluation/Benchmark.tex

```diff
@@ -0,0 +1,202 @@
+We expect the response time to increase linearly with the number of requests as only one request is handled at a time.
+Figure \ref{fig:baseline} shows that this is indeed the case.
+The results for the benchmark of the baseline were examined further to see whether some code submissions were slower to process than others (see \ref{chap:AdditionalPlots}). 
+We found that only one code submission could be considered a direct outlier, namely the one containing syntactical errors.
+This submission was faster to process, likely due to the exercise solution never being executed.
+
+
+\subsection{Queue system}
+\begin{figure}[!tbp]
+  \begin{minipage}[t]{0.4\textwidth}
+    \centering
+    \includegraphics[scale=0.65]{images/10.png}
+    \caption{Mean response time for handling 10 simultaneous client requests for different polling times and Test Runner configurations}
+    \label{fig:resultstart}
+  \end{minipage}
+  \hfill
+  \begin{minipage}[t]{0.4\textwidth}
+    \centering
+    \includegraphics[scale=0.65]{images/20.png}
+    \caption{Mean response time for handling 20 simultaneous client requests for different polling times and Test Runner configurations}
+  \end{minipage}
+  \begin{minipage}[t]{0.4\textwidth}
+    \centering
+    \includegraphics[scale=0.65]{images/50.png}
+    \caption{Mean response time for handling 50 simultaneous client requests for different polling times and Test Runner configurations}
+  \end{minipage}
+  \hfill
+  \begin{minipage}[t]{0.4\textwidth}
+    \centering
+    \includegraphics[scale=0.65]{images/100.png}
+    \caption{Mean response time for handling 100 simultaneous client requests for different polling times and Test Runner configurations}
+    \label{fig:resultEnd}
+  \end{minipage}
+\end{figure}
+
+Figures \ref{fig:resultstart} --- \ref{fig:resultEnd} shows the mean response time of benchmarking the asynchronous Test Runner with different queue sizes, number of threads, and with clients having varying polling times. 
+For each plot, line T denotes the maximum number of threads allowed for testing client submissions, and C denotes the maximum capacity of the channels. 
+All the benchmarks shown in figures \ref{fig:resultstart} --- \ref{fig:resultEnd} sweep the result queue every ten seconds, and remove Test Runner results older than 90 seconds. 
+We see that a Test Runner with Queue System configured to have a maximum capacity of ten and a maximum of five threads surpasses all other configurations in terms of speed, regardless of the client pull time. 
+Compared to the baseline shown on figure \ref{fig:baseline}, it is vastly superior, requiring only a third of the time responding to 100 simultaneous requests. 
+Since a configuration of a Test Runner with five threads and with a channel capacity of ten yields such good results, we examine similar configurations to find a closer estimate to optimal configuration for this particular computer. 
```

Comment:
```suggestion
Since the configuration of the asynchronous Test Runner with five threads and with a channel capacity of ten yields such good results, we examined similar configurations to find a closer estimate to optimal configuration for this particular computer. 
```